urdf_usd_converter/_impl/link.py

```diff
@@ -1,10 +1,295 @@
+        """
+        Get the children of a link.
+        """
+        if link_name not in self.link_tree:
+            return []
+        return self.link_tree[link_name]["children"]
+
+    def get_link_by_name(self, link_name: str) -> ElementLink:
+        """
+        Get a link by name.
+        """
+        return next((link for link in self.root_element.links if link.name == link_name), None)
+
+
+def convert_link(parent: Usd.Prim, link_hierarchy: LinkHierarchy, link: ElementLink, data: ConversionData) -> UsdGeom.Xform:
+    link_safe_name = data.name_cache.getPrimName(parent, link.name)
+    link_xform = usdex.core.defineXform(parent, link_safe_name)
+    link_prim = link_xform.GetPrim()
+    if link.name != link_safe_name:
+        usdex.core.setDisplayName(link_prim, link.name)
+
+    data.references[Tokens.Physics][link.name] = link_prim
+
+    apply_physics_rigidbody(link_prim, data)
+
+    # Assigning MassAPI to a Rigid Body.
+    apply_inertial(link_prim, link, data)
+
+    # Create visual or collision geometry.
+    geometry_basses: list[ElementVisual | ElementCollision] = [visual for visual in link.visuals if visual.geometry and visual.geometry.shape] + [
+        collision for collision in link.collisions if collision.geometry and collision.geometry.shape
+    ]
+
+    for geometry_base in geometry_basses:
+        name = get_geometry_name(geometry_base)
+        geom_safe_name = data.name_cache.getPrimName(link_prim, name)
+        geom_prim = convert_geometry(link_prim, geom_safe_name, geometry_base.geometry, data)
+        if geom_prim:
+            is_collision = isinstance(geometry_base, ElementCollision)
+            if name != geom_safe_name:
+                usdex.core.setDisplayName(geom_prim.GetPrim(), name)
+            set_transform(geom_prim, geometry_base)
+            if is_collision:
+                geom_prim.GetPurposeAttr().Set(UsdGeom.Tokens.guide)
+
+            # Set the physics rigidbody and collision for the geometry.
+            if is_collision:
+                # Apply CollisionAPI to collision geometry
+                apply_physics_collision_mesh(geom_prim.GetPrim(), data)
+
+    children = link_hierarchy.get_link_children(link.name)
+    joints = link_hierarchy.get_link_joints(link.name)
+
+    if len(children) > 0:
+        for child, joint in zip(children, joints):
+            child_xform = convert_link(link_prim, link_hierarchy, child, data)
+            set_transform(child_xform, joint)
+
+    return link_xform
+
+
+def apply_physics_rigidbody(prim: Usd.Prim, data: ConversionData):
+    """
+    Apply the physics rigidbody to a prim.
+    """
+    prim_over = data.content[Tokens.Physics].OverridePrim(prim.GetPath())
+    UsdPhysics.RigidBodyAPI.Apply(prim_over)
+
+
+def apply_physics_collision_mesh(geom_prim: Usd.Prim, data: ConversionData):
+    """
+    Apply the physics collision to a collision mesh geometry.
+    This applies CollisionAPI and MeshCollisionAPI (for mesh types).
```

Comment:
These must be applied to the Gprim directly. In the case of multiple meshes in obj/dae I think its possible that `geom_prim` here is not a Gprim, is that correct?